### PR TITLE
Log ARM deployment error

### PR DESCRIPTION
**Pull request by fiunchinho**
In order to fix this https://github.com/giantswarm/giantswarm/issues/7051 we need to improve our visibility of what's going on here.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func mainError() error {
 	}
 
 	// We define a server factory to create the custom server once all command
-	// line flags are parsed and all microservice configuration is storted out.
+	// line flags are parsed and all microservice configuration is sorted out.
```
Comment (tuommaki) [inline main.go]: <3

File: service/controller/v11/resource/deployment/resource.go [hunk 2]
```diff
@@ -126,8 +126,9 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	_, err = deploymentsClient.CreateOrUpdateResponder(res.Response())
+	deploymentExtended, err := deploymentsClient.CreateOrUpdateResponder(res.Response())
 	if err != nil {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deployment failed", "deployment", deploymentExtended, "error", err)
```
Comment (tuommaki) [inline service/controller/v11/resource/deployment/resource.go]:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deployment failed; deployment: %#q", deploymentExtended), "stack", microerror.Stack(microerror.Mask(err)))
```

I don't remember if I mentioned it earlier, but we have our own [fmt](https://github.com/giantswarm/fmt) repo with various formatting / style / coding guidelines that is useful to take a look for patterns. Regarding logging, there's also one [page](https://github.com/giantswarm/fmt/blob/master/go/log_messages.md#error-handling).

Note that `LogCtx()` doesn't format its strings/values properly so when dumping variables, it's best to use `fmt.Sprintf()`.

**Comment by fiunchinho** [comment]
Fixed

**Comment by fiunchinho** [comment]
Moved changes to v12. They seem to work as expected.

**Comment by tuommaki** [comment]
> They seem to work as expected.

So does this change now enable automatic reconciliation of failed deployments?

**Comment by fiunchinho** [comment]
Yes, I think so 💃

**Comment by tuommaki** [comment]
That's cool! As a last thing: Add ChangeLog message here: https://github.com/giantswarm/azure-operator/blob/master/service/controller/v12/version_bundle.go#L9

See `fmt` entry here: https://github.com/giantswarm/fmt/blob/master/releases/how_write_changelog_message.md

**Comment by fiunchinho** [comment]
done

**Pull request merged**